src/events/eventActions.js

```diff
@@ -59,17 +61,19 @@ export const startEventPolling = (queueId: number, eventId: number) => async (
 
   // eslint-disable-next-line no-constant-condition
   while (true) {
-    const auth = tryGetAuth(getState());
-    if (!auth) {
-      // User switched accounts or logged out
+    if (!authEquivalent(auth, tryGetAuth(getState()))) {
+      // The user logged out or switched accounts during progressiveTimeout
+      // called from see catch block, below. (If tryGetAuth returns undefined,
+      // the user is not logged in.)
```

Comment:
This would make more sense immediately after the relevant `await`, rather than up here as far removed from it as possible. That wouldn't quite be `[nfc]` on its own, though, since it's entirely possible that this function is itself async-delayed (and whether or not it is is a bit fragile, anyway). You'd have to duplicate it to also occur before loop entry.

I wouldn't ordinarily suggest that... but given the duplicated parenthetical in the comment, perhaps this test should be pulled into a mini-closure anyway?

Reply:
... and also used on `catch` entry. If a user has logged out and into another account, you probably don't want to fire off the `deadQueue()` logic – or, worse yet, `logout()`.

Reply:
> ... and also used on `catch` entry.

Good catch! (No pun intended.)

And I agree these checks are best done immediately after the `await`.

Then one other point which can help keep things reasonably simple when checking this at the top of the `catch`: everything else in that `try` block doesn't need to be there, and can be moved to after the `try .. catch`. That error-handling code (including the backoff) only makes sense for an exception thrown inside the API call.

> it's entirely possible that this function is itself async-delayed (and whether or not it is is a bit fragile, anyway). You'd have to duplicate it to also occur before loop entry.

I think the best fix for this is to move the `tryGetAuth(getState())` from its caller down into the top of this function.


> given the duplicated parenthetical in the comment

I think the parenthetical is best just deleted. Its content is in the summary line of the mentioned function's jsdoc anyway.